### PR TITLE
Add test cases for server connector error scenarioes

### PR DESCRIPTION
**Pull request by chamil321**
## Automation tests
 - Unit tests 
   > N/A
 - Integration tests
   > Included

### CODE REVIEW

File: components/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/util/client/http/HttpClient.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+        EventLoopGroup group = new NioEventLoopGroup();
+        ByteBuf content = null;
+        try {
+            content = Unpooled.wrappedBuffer(stringContent.getBytes("UTF-8"));
+        } catch (UnsupportedEncodingException e) {
+            logger.error("Error occurred during string content processing: ", e);
+        }
+        try {
+            Bootstrap b = new Bootstrap();
```
Comment (shafreenAnfar) [inline components/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/util/client/http/HttpClient.java]: shall we avoid single letter variables.

File: components/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/serverconnectorerror/ServerConnectorErrorListener.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    public ServerConnectorErrorListener(TransportsConfiguration configuration) {
+        this.configuration = configuration;
+    }
+
+    @Override
+    public void onMessage(HTTPCarbonMessage httpRequestMessage) {
+        executor.execute(() -> {
+            HTTPCarbonMessage responseMsg = httpRequestMessage.cloneCarbonMessageWithData();
+            try {
+                Thread.sleep(1000);
```
Comment (shafreenAnfar) [inline components/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/serverconnectorerror/ServerConnectorErrorListener.java]: Shall we add a comment here explaining why we had to add a thread sleep. Having to have a thread sleep is always bad :(

File: components/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/util/client/http/HttpClient.java (hunk 1 above)
Comment (chamil321) [reply components/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/util/client/http/HttpClient.java]: Ack

File: components/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/serverconnectorerror/ServerConnectorErrorListener.java (hunk 2 above)
Comment (chamil321) [reply components/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/serverconnectorerror/ServerConnectorErrorListener.java]: Ack

**Comment by shafreenAnfar** [comment]
We thought of writing this without using thread sleeps. Solution was to write a simple TCP client and get rid of the thread sleep. Hence, I am closing this PR without merging.

**Pull request closed without merging**